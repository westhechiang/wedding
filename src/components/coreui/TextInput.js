import styled from 'styled-components';
import { theme } from '../../ui';

const themes = {
  blueLine: `
    background: transparent;
    border: none;
    border-bottom: 1px solid ${theme.color.cyan4};
    color: ${theme.color.white};
    font-size: ${theme.font.large};

    &:focus {
      outline: none;
    }

    &::placeholder {
      color: ${theme.color.cyan4};
      font-weight: ${theme.font.weight.light};
      font-family: ${theme.font.family};
      font-size: ${theme.font.large};
    }
  `,
  mapSearch: `
    background: ${theme.color.white};
    border: none;
    box-shadow: ${theme.shadow};
    color: ${theme.color.grayExtraDark};
    border-radius: 3px;

    &::placeholder {
      color: ${theme.color.grayExtraDark};
      font-family: ${theme.font.family};
      font-size: ${theme.font.small};
    }
  `,
  deals: `
    border-radius: ${theme.borderRadius};
    width: 300px;
    border: 1px solid ${theme.color.grayExtraLight};
    text-indent: 20px;
  `,
  refineList: `
    background: ${theme.color.grayExtraDark};
    box-shadow: inset 0 1px 2px 2px ${theme.color.grayExtraExtraDark};
    border: none;
    border-bottom: 1px solid ${theme.color.gray};
    height: 20px;
    border-radius: ${theme.borderRadius};
    color: ${theme.color.grayExtraLight};
    text-align: center;
    font-size: ${theme.font.xsmall};

    &::placeholder {
      color: ${theme.color.grayExtraLight};
      font-family: ${theme.font.family};
      font-size: ${theme.font.xsmall};
      font-weight: ${theme.font.weight.light};
    }
  `,
  dealDetails: `
    border-radius: ${theme.borderRadius};
    border: 1px solid ${theme.color.grayExtraLight};
    width: 50%;
    text-align: right;
    background: ${theme.color.grayExtraExtraExtraLight};
  `,
};


const TextInput = styled.input.attrs({
  type: 'text',
})`
  box-sizing: border-box;
  font-family: ${theme.font.family};
  font-size: ${theme.font.small};
  font-weight: ${theme.font.weight.light};
  height: 30px;
  padding: 10px;
  width: 100%;
  display: ${props => props.display || 'inline'};
  text-align: ${props => props.textAlign || 'start'};
  ${props => themes[props.theme]}
`;

export default TextInput;
