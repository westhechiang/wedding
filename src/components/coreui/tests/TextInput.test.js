import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TextInput from '../../../components/coreui/TextInput';

describe('<TextInput />', () => {
  it('should match its empty snapshot', () => {
    const wrapper = shallow(<TextInput />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "blueLine" theme snapshot', () => {
    const wrapper = shallow(<TextInput theme="blueLine" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "mapSearch" theme snapshot', () => {
    const wrapper = shallow(<TextInput theme="mapSearch" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "refineList" theme snapshot', () => {
    const wrapper = shallow(<TextInput theme="refineList" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
