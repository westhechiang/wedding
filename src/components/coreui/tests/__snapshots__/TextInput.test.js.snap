// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match its "blueLine" theme snapshot 1`] = `
<input
  className="sc-bdVaJa khiFTh"
  type="text"
/>
`;

exports[`<TextInput /> should match its "mapSearch" theme snapshot 1`] = `
<input
  className="sc-bdVaJa euRxdZ"
  type="text"
/>
`;

exports[`<TextInput /> should match its "refineList" theme snapshot 1`] = `
<input
  className="sc-bdVaJa jhWEXd"
  type="text"
/>
`;

exports[`<TextInput /> should match its empty snapshot 1`] = `
<input
  className="sc-bdVaJa hLUzGS"
  type="text"
/>
`;
